tests/unit/test_requirements_generation.py

```
@@ -0,0 +1,185 @@
+    )
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_path = Path(temp_dir)
+        pyproject_path = temp_path / "pyproject.toml"
+        requirements_path = temp_path / "requirements.txt"
+
+        # Write test pyproject.toml
+        pyproject_path.write_text(pyproject_content)
+
+        # Import and use our requirements generator (will fail initially)
+        from src.requirements_generator import generate_requirements
```

Ruff (0.12.2)

`import` should be at the top-level of a file

(PLC0415)